metadata-ingestion/tests/unit/test_teradata_source.py

```diff
@@ -1494,3 +1496,267 @@ def _create_mock_entry(
         entry.default_database = default_database
         entry.session_id = session_id or f"session_{query_id}"
         return entry
+
+
+class TestTeradataRegressionFixes:
+    """Test suite for regression fixes in Teradata source.
+
+    This class tests the fixes for two critical issues:
+    1. Time window defaults not being applied correctly
+    2. Missing incremental lineage support
+    """
+
+    def test_time_window_defaults_applied(self):
+        """Test that BaseTimeWindowConfig defaults are automatically applied.
+
+        This test ensures that the fix for the time window issue works correctly.
+        Previously, start_time and end_time were None before pydantic validation,
+        causing malformed SQL queries.
+        """
+        # Configuration without explicit time ranges (user's original recipe scenario)
+        config_dict = {
+            "username": "test_user",
+            "password": "test_password",
+            "host_port": "localhost:1025",
+            "include_table_lineage": True,
+            "include_usage_statistics": True,
+        }
+
+        config = TeradataConfig.parse_obj(config_dict)
+
+        # Verify that BaseTimeWindowConfig defaults are applied
+        assert config.start_time is not None, (
+            "start_time should not be None after pydantic validation"
+        )
+        assert config.end_time is not None, (
+            "end_time should not be None after pydantic validation"
+        )
+
+        # Verify they are datetime objects
+        assert isinstance(config.start_time, datetime), (
+            "start_time should be a datetime object"
+        )
+        assert isinstance(config.end_time, datetime), (
+            "end_time should be a datetime object"
+        )
+
+        # Verify reasonable defaults (end_time should be after start_time)
+        assert config.end_time > config.start_time, (
+            "end_time should be after start_time"
+        )
+
+        # Verify they can be used in SQL queries (both formats work fine)
+        # Test both formatting approaches work
+        formatted_str = config.start_time.strftime("%Y-%m-%d %H:%M:%S")
+        direct_str = str(config.start_time)
+
+        assert len(formatted_str) > 0, "strftime formatting should work"
+        assert len(direct_str) > 0, "direct string conversion should work"
+        assert "None" not in formatted_str, "formatted time should not contain 'None'"
+        assert "None" not in direct_str, "direct string should not contain 'None'"
```

Comment:
this feels like we are testing formatting methods in `datetime` class itself, is that necessary?